storage/integration_test.go

```diff
@@ -659,6 +657,14 @@ func TestIntegration_BucketPolicyOnly(t *testing.T) {
 	}
 }
 
+func retryOnNilAndTransientErrs(err error) bool {
+	return err == nil || ShouldRetry(err)
+}
+func retryOn400AndTransientErrs(err error) bool {
+	var e *googleapi.Error
+	return ShouldRetry(err) || errors.As(err, &e) && e.Code == 400
```

Comment:
Do we see any of these tests failing with other codes (e.g. 401 or 403)? Would be good to double check.